### PR TITLE
Fix VisibleDeprecationWarning

### PR DESCRIPTION
**Pull request by shaynekang**
XGBoost raises VisibleDeprecationWarning when the numpy version is over 0.9.

```
(xgboost-3.5.1) SUNGHEEui-MacBook:xgboost shaynekang$ nosetests tests/python/

...

/Users/shaynekang/.pyenv/versions/3.5.1/envs/xgboost-3.5.1/lib/python3.5/site-packages/xgboost-0.4-py3.5.egg/xgboost/training.py:273: VisibleDeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  idset = [randidx[(i * kstep): min(len(randidx), (i + 1) * kstep)] for i in range(nfold)]

...

/Users/shaynekang/.pyenv/versions/3.5.1/envs/xgboost-3.5.1/lib/python3.5/site-packages/xgboost-0.4-py3.5.egg/xgboost/core.py:907: VisibleDeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
  preds = preds.reshape(nrow, preds.size / nrow)

...

----------------------------------------------------------------------
Ran 31 tests in 1.260s
```

All of these problems cause misuse of datatype. I try to convert non-integer variables to integer than problems are solved.

### CODE REVIEW

File: python-package/xgboost/training.py [hunk 1]
```diff
@@ -270,6 +270,7 @@ def mknfold(dall, nfold, param, seed, evals=(), fpreproc=None, stratified=False,
     if stratified is False and folds is None:
         randidx = np.random.permutation(dall.num_row())
         kstep = len(randidx) / nfold
+        kstep = int(kstep)
```
Comment (terrytangyuan) [inline python-package/xgboost/training.py]: You can just call int on the first kstep assignment

**Comment by terrytangyuan** [comment]
Please rebase with master

**Comment by shaynekang** [comment]
Done. Is there any other things that I can help with?

**Comment by tqchen** [comment]
seems the rebase trace was somewhat mixed with exising trace. We might need to ask you start from a new clean fork

**Comment by shaynekang** [comment]
Sure. May I create another pull request from master, or any other suggestion?

**Comment by tqchen** [comment]
Yes, please create another PR

**Comment by shaynekang** [comment]
see #1181

**Pull request closed without merging**